Source/JavaScriptCore/runtime/JSArray.cpp

```diff
@@ -1023,6 +1023,178 @@ JSString* JSArray::fastToString(JSGlobalObject* globalObject)
     RELEASE_AND_RETURN(scope, joiner.join(globalObject));
 }
 
+static uint64_t calculateFlattenedLength(JSArray* sourceArray, uint64_t sourceLength, uint64_t depth)
+{
+    uint64_t resultLength = 0;
+    IndexingType sourceType = sourceArray->indexingType();
+
```

Comment:
We need to have recursion check via `vm.isSafeToRecurseSoft`. `flatIntoArray` is recursively calling, so it can be possible that we will have stackoverflow.